src/core/selection/selection.view.js

```diff
@@ -2,43 +2,243 @@ import View from '../view/view';
 import Command from 'core/infrastructure/command';
 import behaviorFactory from './selection.factory';
```

Comment:
As strategy is a part of behavior algorithms it can be renamed, but from readable context i like behavior more here